src/components/Download.tsx

```diff
@@ -1,10 +1,232 @@
+import { useState, useMemo } from 'react';
+import {
+  Button,
+  Alert,
+  Collapse,
+  Typography,
+  List,
+  ListItem,
+  Switch,
+  FormControlLabel,
+  Link,
+} from '@mui/material';
+import Ajv from 'ajv';
 import NavigationButton from './NavigationButton';
+import DataDictionaryPreview from './DataDictionaryPreview';
+import useDataStore from '../stores/data';
+import schema from '../assets/neurobagel_data_dictionary.schema.json';
+import { DataDictionary } from '../utils/types';
+import emoji from '../assets/download-emoji.png';
 
 function Download() {
+  const [dictionaryCollapsed, setDictionaryCollapsed] = useState(false);
+  const [forceAllowDownload, setForceAllowDownload] = useState(false);
+
+  const uploadedDataTableFileName = useDataStore((state) => state.uploadedDataTableFileName);
+
+  const columns = useDataStore((state) => state.columns);
+
+  const dataDictionary = useMemo(
+    () =>
+      Object.entries(columns).reduce((acc, [_, column]) => {
+        if (column.header) {
+          acc[column.header] = {
+            Description: column.description || '',
+          };
+        }
+        return acc;
+      }, {} as DataDictionary),
+    [columns]
+  );
+
+  const { isValid: schemaValid, errors: schemaErrors } = useMemo(() => {
+    const ajv = new Ajv();
+    const validate = ajv.compile(schema);
+    const isValid = validate(dataDictionary);
+
+    if (!isValid) {
+      const errors = validate.errors?.map((error) => error.instancePath.slice(1)) || [];
```

Comment:
you may want to add a comment here on why you are taking a slice on the instancepaths, i.e. why do we care to remove the slash?